caas/docker/images.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/juju/errors"
+	"github.com/juju/version"
+)
+
+const (
+	dockerOrgName   = "ycliuhw"
+	dockerNameSpace = "jujud-controller"
+)
+
+// NewClient constructs a new docker client.
+func NewClient() (*client.Client, error) {
+	return client.NewEnvClient()
+}
+
+// InspectImage inspects image and returns the image information and its raw representation.
+func InspectImage(c client.ImageAPIClient, imagePath string) (types.ImageInspect, error) {
+	o, _, err := c.ImageInspectWithRaw(context.Background(), imagePath)
+	return o, err
+}
+
+// PullImage pulls docker image to local.
+func PullImage(c client.ImageAPIClient, imagePath string) error {
+	reader, err := c.ImagePull(context.Background(), "docker.io/library/alpine", types.ImagePullOptions{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	io.Copy(os.Stdout, reader)
+	return nil
+}
+
+func jujuVersionToDockerImagePath(v string) string {
+	return fmt.Sprintf("%s/%s:%s", dockerOrgName, dockerNameSpace, v)
+}
+
+// GetToolImagePath returns a jujud docker image path for the specified juju version.
+func GetToolImagePath(c client.ImageAPIClient, toolVersion version.Number) (string, error) {
```

Comment:
We don't want any of this - as with the operator images, we tie a specific version of Juju to a tagged docker image. We don't query the repo. When Juju is built, we upload the matching image, tagged with the Juju version string. So all we need to do is make the podspec image path point to the right location.

Reply:
removed